File: BukkitUtil-core/src/main/java/no/kh498/util/jackson/deserializers/key/EnchantmentKeyDeserializer.java
package no.kh498.util.jackson.deserializers.key;

import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.KeyDeserializer;
import org.bukkit.enchantments.Enchantment;

/**
 * @author Elg
 */
public class EnchantmentKeyDeserializer extends KeyDeserializer {

    static {
        inst = new EnchantmentKeyDeserializer();
    }

    public static final EnchantmentKeyDeserializer inst;

    private EnchantmentKeyDeserializer() { }

    @Override
    public Object deserializeKey(String key, DeserializationContext ctxt) {
        int id;
        try {
            id = Integer.parseInt(key);
        } catch (NumberFormatException e) {
            throw new IllegalStateException("Expected to find an integer but found " + key);
        }
        //noinspection deprecation
        return Enchantment.getById(id);
    }
}

File: BukkitUtil-core/src/main/kotlin/no/kh498/util/jackson/deserializers/key/EnchantmentKeyDeserializer.kt
package no.kh498.util.jackson.deserializers.key

import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.KeyDeserializer
import org.bukkit.enchantments.Enchantment

/**
 * @author Elg
 */
object EnchantmentKeyDeserializer : KeyDeserializer() {

    override fun deserializeKey(key: String, ctxt: DeserializationContext): Any {
        val id: Int
        id = try {
            key.toInt()
        } catch (e: NumberFormatException) {
            throw IllegalStateException("Expected to find an integer but found $key")
        }
        return Enchantment.getById(id)
    }
}
